src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -771,10 +791,13 @@ private IOperation CreateBoundConversionOperation(BoundConversion boundConversio
                     // and instead directly return the operand, which will be a BoundBadExpression. When we generate a node for the BoundBadExpression,
                     // the resulting IOperation will also have a null Type.
                     Debug.Assert(boundConversion.Operand.Kind == BoundKind.BadExpression);
-                    return Create(boundConversion.Operand);
+                    
+                    // We invoke CreateInternalCore for implicit conversions to avoid duplicate IParenthesizedOperation for bound conversion and its operand.
+                    return isImplicit ? CreateInternalCore(boundConversion.Operand) : Create(boundConversion.Operand);
```

Comment:
This and the lambda change removes things from the cache. Perhaps just give Create an optional argument supressParenthesizedCreation, so we can keep the same code path and ensure everything remains in the cache?

Reply:
The parent should still be in the cache and given that GetOperation API anyway walks up the chain to the binding root (statement), we should be fine.